tests/stdlib/tosenv.nim

```diff
@@ -0,0 +1,40 @@
+discard """
+  matrix: "--threads"
+  joinable: false
+  targets: "c cpp"
+"""
+import os
+
+block delEnv:
+    const dummyEnvVar = "D20210720T144752" # This env var wouldn't be likely to exist to begin with
+    doAssert (not existsEnv(dummyEnvVar))
+    putEnv(dummyEnvVar, "1")
+    doAssert existsEnv(dummyEnvVar) 
+    delEnv(dummyEnvVar)
+    doAssert (not existsEnv(dummyEnvVar))
+    delEnv(dummyEnvVar) # deleting an already deleted env var
+    doAssert (not existsEnv(dummyEnvVar))
+
+block putEnv:
+    # raises OSError on invalid input
+    doAssertRaises(OSError, putEnv("DUMMY_ENV_VAR_PUT=DUMMY_VALUE",
+            "NEW_DUMMY_VALUE"))
+    doAssertRaises(OSError, putEnv("", "NEW_DUMMY_VALUE"))
```

Comment:
```suggestion
    doAssert not existsEnv("")
    doAssert not existsEnv("DUMMY_ENV_VAR_PUT=DUMMY_VALUE")
    doAssert not existsEnv("DUMMY_ENV_VAR_PUT")
```

Reply:
Are you sure about this? I added those tests to specifically test the behavior of `putEnv` (and the fixing of #18533 )

Reply:
yes, i tried adding these and the tests pass; it ensures that putEnv with invalid values not only raises (as tested) but also doesn't add an environment variable (as these extra lines demonstrate)